java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.formatNumber;
+
+import org.junit.Test;
+
+import static com.baeldung.formatNumber.FormatNumber.*;
+import static org.assertj.core.api.Assertions.*;
+
+public class FormatNumberUnitTest {
+        private static final double D = 4.2352989244d;
+        private static final double F = 8.6994540927d;
+
+        @Test public void givenDecimalNumber_whenFormatNumberToNDecimalPlaces_thenGetExpectedResult() {
+
+                assertThat(withBigDecimal(D, 2)).isEqualTo(4.24);
+                assertThat(withBigDecimal(D, 3)).isEqualTo(4.235);
+                assertThat(withBigDecimal(F, 2)).isEqualTo(8.7);
+                assertThat(withBigDecimal(F, 3)).isEqualTo(8.699);
+
+                assertThat(withMathRound(D, 2)).isEqualTo(4.24);
+                assertThat(withMathRound(D, 3)).isEqualTo(4.235);
+                assertThat(withMathRound(F, 2)).isEqualTo(8.7);
+                assertThat(withMathRound(F, 3)).isEqualTo(8.699);
+
```

Comment:
We should split up the tests into separate `@Test` methods for each solution.  For example, there should be a `@Test` method for `withBigDecimal`, `withMathRound`, `withStringFormat` , `withDecimalFormatLocal` , and`withDecimalFormatPattern` .

Reply:
Okay